webgpu.h

```diff
@@ -144,11 +144,27 @@ typedef enum WGPUCullMode {
     WGPUCullMode_Force32 = 0x7FFFFFFF
 } WGPUCullMode;
 
+typedef enum WGPUErrorFilter {
+    WGPUErrorFilter_None = 0x00000000,
+    WGPUErrorFilter_Validation = 0x00000001,
+    WGPUErrorFilter_OutOfMemory = 0x00000002,
+    WGPUErrorFilter_Force32 = 0x7FFFFFFF
```

Comment:
These `Force32` things are quite unsound. Can we remove them?

Reply:
Hm, I thought this happened already in the original PR.

Doing this would mean splitting off the typedefs from the enums:

```c
typedef uint32_t WPUEnum;
```
```c
enum {
    WGPUErrorFilter_None = 0x00000000,
    WGPUErrorFilter_Validation = 0x00000001,
    WGPUErrorFilter_OutOfMemory = 0x00000002,
    WGPUErrorFilter_Force32 = 0x7FFFFFFF
}
typedef WGPUEnum WGPUErrorFilter;
```

, which looks OK to me.

Better to do it in another PR though.

Reply:
It doesn't look like `Force32` gives you what you need anyway. It doesn't guarantee that the compiler chooses 32 bits here.
I agree about discussing it in a different PR though

Reply:
Whoops somehow I thought I had fixed flags to be to something we agreed to. I must have messed something up. I'll revisit that change.

Reply:
Ah, you'd like this for enums as well? I don't think that's a great idea because it loses even more typing information that's useful for code completion for example.

Reply:
In C I don't think there is actually any typing information here: https://godbolt.org/z/rwMsAV
but maybe some autocompletion engines would use it anyway.

Reply:
I'm not following. Removing "Force32" loses even more typing?

Reply:
I think the typing loss comes from changing the type of WGPUErrorFilter from `enum WGPUErrorFilter` to `uint32_t`.

Reply:
Sad indeed, although:
  1. you are going to be mostly using the API via a C++ wrapper, it's just unfortunate to require to to duplicate all the enums
  2. how else would we be able to force it to 32 bits?

Reply:
`Force32` doesn't guarantee 32bit but that's the mechanism Vulkan uses already. I'll put up a PR and we can ask some folks that would use WebGPU in C (@floooh ?)

Reply:
oh snap. So all vulkan headers are technically UB in terms of enum usage ? yikes